### ISSUE TITLE
RFC: Enabling GraphQL endpoint to support special characters in column names

### ISSUE THREAD

**Issue by vadeveka** [Azure/data-api-builder#1860]
A table or a view in SQL may have special characters such as space for column names. If the developer does not provide explicit fieldmappings under the entity in config, then the field names are derived as is from column names. However special characters such as spaces are not supported in GraphQL names([acceptable characters per spec](https://spec.graphql.org/October2021/#sec-Names)) and while DAB starts up fine, the GraphQL calls fail due to invalid schema.

See related discussion on #1528, #692 and #1479. Note that the issue does not impact REST endpoint/swagger as special characters for entity fields are escaped and accepted.

There are several ways we can look to address the issue for GraphQL involved if no field mappings provided:
The approaches take effect only when GraphQL endpoint is enabled.
### Throw error upfront
- During GraphQL schema generation, DAB will verify that column names are valid to be used as GraphQL field names.
- If invalid, then throw fatal error during startup/config enablement and provide suggestion to use fieldmappings in config.

### Introduce config option - called sanitize-graph-fields with boolean type defaulting to false.
- In this, DAB will verify that column names are valid to be used as GraphQL field names. 
- If invalid and sanitize-graph-fields = false, then throw error upfront similar to previous approach. 
- If invalid and sanitize-graph-fields = true, then sanitize the column names to field names using [this styling convention](https://github.com/hendrikniemann/graphql-style-guide#fields) (already being leveraged in DAB for query names). Alternately we can choose to always sanitize all field names per the convention even if column names are valid for GraphQL naming.
   - Note that the sanitized field names will be used for both GraphQL and REST endpoint for parity across both endpoints and to avoid introducing complexity with maintaining separate mappings

**Comment by vadeveka** [comment]
Option 2 originally seeks to apply the naming convention only if characters are invalid for GraphQL naming. This means that if the names are valid (e.g. _myType) but not necessarily following the convention (recommends camcelcase and no underscore), there would be no sanitization applied. 

The alternate proposal was to always sanitize the field names if not following convention. I dont suggest we do this although some may prefer that `sanitize-fields` should always enforce naming convention for clarity.

`sanitize-fields` as the option name sounds fine to go with.

**Comment by seantleonard** [comment]
Is #1535 covered within this scope?

**Comment by vadeveka** [comment]
Good point. If the names are non-English characters, that may cause an impact to name sanitization.
Added below proposal:

> Given GraphQL names can only be in English chars, if the name sanitization results in only underscores left, then throw error indicating that valid field names could not be derived and to use fieldmappings.

**Comment by seantleonard** [comment]
The word `sanitize` doesn't make sense for REST. Because REST doesn't require field name changes and GraphQL does, I would argue for a property name using the word "standard" like `standardize-fieldnames`.
- GraphQL: align field names with GraphQL specification to enable scheme building success.
- REST: 'standardize' field names to align with names used for GraphQL.

**Comment by vadeveka** [comment]
fair point - updated proposal to change config option name
